File: design-patterns/go/adapter/transport/plane.go
// plane.go

package main

type Plane struct {
}

func NewPlane() (plane *Plane) {
	plane = &Plane{}
	return
}

func (plane *Plane) GetNumberOfEngines() int {
	return 2
}

func (plane *Plane) GetNumberOfWheels() int {
	return 3
}

func (plane *Plane) GetWeight() float64 {
	return 127_000
}

func (plane *Plane) GetDistanceTravelled() float64 {
	return 500 // Nautical miles
}

func (plane *Plane) GetTravelCostTotal() float64 {
	return 3_000
}

File: design-patterns/ts/adapter/transport/plane.ts
// plane.ts

import AirTransport from "./air-transport";

class Plane implements AirTransport {
    getNumberOfWheels(): number {
        return 3;
    }
    
    getNumberOfEngines(): number {
        return 2;
    }

    getWeight(): number {
        return 127000;
    }

    getDistanceTravelled(): number {
        return 500;
    }

    getTravelCostTotal(): number {
        return 3000;
    }
}

export default Plane;